token/client/src/token.rs

```diff
@@ -2261,6 +2263,33 @@ where
             )
         };
 
+        let (transfer_amount_auditor_ciphertext_lo, transfer_amount_auditor_ciphertext_hi) =
+            if let Some(proof_data) = ciphertext_validity_proof_data {
+                let transfer_amount_auditor_ciphertext_lo = proof_data
+                    .context_data()
+                    .grouped_ciphertext_lo
+                    .try_extract_ciphertext(2)
+                    .map_err(|_| TokenError::ProofGeneration)?;
+                let transfer_amount_auditor_ciphertext_hi = proof_data
+                    .context_data()
+                    .grouped_ciphertext_hi
+                    .try_extract_ciphertext(2)
+                    .map_err(|_| TokenError::ProofGeneration)?;
```

Comment:
If I've understood this correctly, the proof data is created in this function, right before, so would it be possible for this error to happen ever? We might prefer an `unwrap()` in that case. Or maybe even simpler, we can return them from the earlier block that generates the proof data